### PR TITLE
Add support for output formats, including a new, experimental 'simplified-rst' format

### LINKED ISSUE #165: Make RST output configurable
Chose between consumption by Sphinx and consumption by ... others? What exactly this others is needs to be defined, but a big use-case would be things like https://github.com/ansible-collections/community.vmware/issues/1770.

### PR DESCRIPTION
**Pull request by felixfontein**
Fixes #165.

CC @mariolenz  - this is probably not yet exactly what you need, but a first step in that direction. I've labelled the format experimental so we can do big changes to it without having to release new major versions, until we declare it stable :)

### CODE REVIEW

**Comment by felixfontein** [comment]
You can see the result rendered on GitHub here:

https://github.com/felixfontein/antsibull-docs/blob/formats/tests/functional/baseline-simplified-rst-squash-hierarchy/index.rst

https://github.com/felixfontein/antsibull-docs/blob/formats/tests/functional/baseline-simplified-rst/collections/index.rst

Please note that some of the links are intentionally broken. Something that's also strange is that GitHub isn't rendering the `list-table` tables. In principle these should work, see https://gist.github.com/felixfontein/b8092412d8e6d900a99a3c7e14453a89 - I have to figure out what goes wrong...

**Comment by felixfontein** [comment]
The problem seems to be `:width: 100%` - once that is present, GitHub completely ignores the tables...

**Comment by felixfontein** [comment]
Now it works :)

**Comment by felixfontein** [comment]
Results from https://github.com/ansible-collections/community.vmware/pull/1788:
- `.. note::` does not work with GH
- `yaml+jinja2` results in no syntax highlighting on GH; better stick to `yaml`
- comparing https://github.com/felixfontein/community.vmware/blob/antsibull-docs/docs/community.vmware.vmware_vcenter_settings_module.rst#parameters with https://github.com/felixfontein/community.vmware/blob/main/docs/community.vmware.vmware_vcenter_settings_module.rst#parameters shows that suboptions do not work (in the sense that they do not render correctly) on GH

**Comment by felixfontein** [comment]
Hmm, I'm afraid that if we want to use tables for options with proper suboptions support, we'll have to use HTML tables instead of RST tables. (That's also what collection_prep is doing.) Any thoughts on this?

**Comment by felixfontein** [comment]
I switched to RST tables instead of HTML tables, and now it looks pretty well. (Also note that GitHub strips all `style` attributes when rendering.)

**Comment by felixfontein** [comment]
I've declared https://github.com/ansible-collections/community.vmware/pull/1788 ready for review, which also means that I think this PR is done.

**Comment by felixfontein** [comment]
@gotmax23 @mariolenz any objections to merging this soon? I'd like to get this out into a release so it's easier to try this out. IMO the result looks quite ok at the moment.

**Review by gotmax23 — COMMENTED** [review]
I gave this a quick look through. I think we can ship it.

File: src/antsibull_docs/cli/doc_commands/collection_plugins.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        collection_tarballs = asyncio.run(
+            retrieve(
+                list(app_ctx.extra["collection"]),
+                collection_version,
+                tmp_dir,
+                galaxy_server=app_ctx.galaxy_url,
+                collection_cache=app_ctx.collection_cache,
+            )
+        )
+        # flog.fields(tarballs=collection_tarballs).debug('Download complete')
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply src/antsibull_docs/cli/doc_commands/collection_plugins.py]: I've uncommented (and also in three other places).

**Comment by felixfontein** [comment]
@mariolenz @gotmax23 thanks!

**Pull request merged**